### ISSUE TITLE
Adapt ARTIK IDE to use Che Docker Launcher

### ISSUE THREAD

**Issue by DRuffer** [eclipse-che/che#2117]
So far, all I see is the Vagrant instructions using VirtualBox, but I have too many things already working in the latest Docker using Hyper-V. Switching back and forth is problematic.

I am able to run both eclipse and codenvy's version of che with:
docker run --rm -t -v /var/run/docker.sock:/var/run/docker.sock codenvy/che start
See https://www.eclipse.org/che/getting-started/download/

I want https://eclipse-che.readme.io/docs/samsung-artik-ide-getting-started
without VirtualBox and Vagrant.

**Comment by TylerJewell** [comment]
Hi @DRuffer - I think what you are asking about is our new Docker Launcher for Eclipse Che 4.6.2. It uses pure Docker capabilities and therefore runs on Windows 10 with Docker for Windows.  You can also use it the same way with boot2docker through VirtualBox.

We have an engineering task to convert the new Docker Launcher for Che to support ARTIK IDE as well - it's probably not that far away.  But we need to do an adaptation of the launcher and then run a battery of tests.

**Comment by DRuffer** [comment]
Yes, I couldn't figure out its name, but I love its performance.

Is there a timeline for the ARTIK IDE?

FYI, I work as a Firmware Engineer within the group (ID8) in the Samsung Strategy and Innovation Center that is creating apps for the ARTIK chips. My email over there is d.ruffer@partner.samsung.com. I suspect that I can't help much over here, but I do have access to the systems and I can test.

**Comment by TylerJewell** [comment]
A major release of the Artik IDE is planned for Monday.  It has a docker container that can be used to run it, but the syntax is complex and different for each OS.  The docker launcher is a way to run the same syntax on any OS.  We don't have a time frame for the launcher et but will get the server syntax posted.

**Comment by DRuffer** [comment]
Thanks!

**Comment by TylerJewell** [comment]
If you want to be a bit more adventurous - there is a container for ARTIK already.  We launch it inside the vagrantfile.  It is here:
https://github.com/codenvy/artik-ide/blob/master/Vagrantfile#L175-L182

**Comment by DRuffer** [comment]
Unfortunately, when I have tried to switch back to Vagrant (e.g. turn off Hyper-V and use VirtualBox) it is real painful to get Docker working back on Hyper-V again. I have the Vagrant tool chain installed, but choose not to go through that pain. Since my primary job is on Artik0, I don't need Che to get my work done. ;)

**Comment by TylerJewell** [comment]
But if you are on Windows 10 with the proper install of docker for Windows, then docker should just work for you right away and you can use the highlighted rows in the vagrantfile to launch natively with docker for Windows.

**Comment by DRuffer** [comment]
I don't know how to do that yet. I tried something like that earlier, and got frustrated. I'll try that again in a few minutes.

**Comment by DRuffer** [comment]
Ok, something happened, but I'm too new to all of this, and way out of my comfort zone, but I got:

> > ```
> >           `codenvy/artikide:latest --remote:192.168.28.200 --port:8080 run
> > ```
> > 
> > 0df4aceab8c112786888bf63fe83270c52e245e9fd8bf05b8abeb662c6c93b7e

and I see that something is running:

PS C:\Users\druff> docker ps
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS
 PORTS               NAMES
0df4aceab8c1        codenvy/artikide:latest   "/home/user/che/bin/c"   2 minutes ago       Restarting (0) 49 seconds ago
                     che

But http://192.168.28.200:8080/ gives me 192.168.28.200 took too long to respond.

**Comment by TylerJewell** [comment]
Yeah - that means the server didn't start.  But it got close.  This just means that it is a configuration issue.  Though without getting hands on it may be difficult to tell what needs to change.  The launcher for Artik should address this more easily.

**Comment by DRuffer** [comment]
I can wait for you all to get it closer. As I said, this is not in my critical path yet.

However, I do sincerely appreciate the time you took to try.

**Comment by TylerJewell** [comment]
Update here.  This has been my side project for the week and I happy to say that it is almost done.  Hopefully a pull request by the weekend.  It runs so much better with the docker launcher.

**Comment by TylerJewell** [comment]
https://github.com/codenvy/artik-ide/pull/117 - this is the pull request that needs to get merged for this to work.  This pull request is also dependent upon some changes being incorporated into Che for the 4.7 release.  Those are getting QA'd and tested.

If you were super ambitious, you could test the new simplified capabilities.  But you'd have to be really motivated to do it.

```
# Create local version of the che-launcher
git clone http://github.com/eclipse/che
git checkout CHE-2116
cd /dockerfiles/che-launcher
bash build.sh

# Create a profile from the new Che CLI
# Save [this file](https://github.com/eclipse/che/blob/66c89828f165cb7ac05fbc94260c5fd4c7b8bb54/che.sh) as che.sh
export CHE_SERVER_IMAGE_NAME=codenvy/artikide
export CHE_SERVER_CONTAINER_NAME=artik-ide
export CHE_VERSION=nightly

./che.sh profile add artik
./che.sh profile set artik
./che.sh profile info artik

# Start Che with the new launcher + artik container.
./che.sh start
```

**Comment by TylerJewell** [comment]
@eivantsov - let's do a test on the new getting started page and using the che-launcher with artik IDE.  I believe all necessary changes have been incorporated and we are ready to go live - and discontinue the vagrant installer.

**Comment by JamesDrummond** [comment]
@TylerJewell Was this completed with the CLI or were we trying to create something like che-launcher for artik that does not require the CLI?

**Comment by TylerJewell** [comment]
Updates to Che launcher and cli allow this to work now.

**Comment by JamesDrummond** [comment]
@DRuffer Have you had a chance to test this to resolve this issue?

**Comment by TylerJewell** [comment]
Closing this issue as it was resolved in 4.7.2.
